### DISCUSSION TITLE
concurrent inserts to different tables in the same file

### DISCUSSION THREAD

**Discussion by drew2323** [msiemens/tinydb#526] [Q&A] [answered]
I have few parallel threads running and inserting data into their own table in tinydb, but sharing the same json db file. Is this supported scenario?

**Comment by MrPigss** [comment, marked as answer]
See this page on the docs: [Why Not Use TinyDB](https://tinydb.readthedocs.io/en/latest/intro.html#why-not-use-tinydb).
One of the first things it says is:
> You need advanced features like:
> - access from multiple processes or threads

This doesn't mean it's not possible but you'll have to make sure you use [locks](https://docs.python.org/3/library/threading.html#lock-objects) appropriately.
